### PR TITLE
Refactoring PsBasic.java

### PR DESCRIPTION
**Pull request by sebing**
Removed the code from the constructor to satisfy ConstructorOnlyInitializesOrCallOtherConstructors errors.

Fixing(https://github.com/yegor256/takes/issues/600)

### CODE REVIEW

**Comment by sebing** [comment]
@davvd Can you review the PR?

**Comment by davvd** [comment]
@sebing Thanks, I will ask someone to do a code review here, thanks for your pull request

**Comment by davvd** [comment]
@mkordas please review, thanks

**Comment by mkordas** [comment]
@sebing I'll do review today

File: src/main/java/org/takes/facets/auth/PsBasic.java [hunk 1]
```diff
@@ -52,7 +52,14 @@
...
  * @version $Id$
  * @since 0.20
+ * @todo #600:30min This class has too many methods. Refactor the class.
+ *  Can move the subclasses into separate classes.
+ *  Remove the warning suppression for `TooManyMethods`.
  */
+@SuppressWarnings
+    (
+        "PMD.TooManyMethods"
+    )
```
Comment (mkordas) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @sebing please join lines here

File: src/main/java/org/takes/facets/auth/PsBasic.java [hunk 2]
```diff
@@ -269,6 +259,27 @@ public Default(final String... users) {
         }
 
         /**
+         * Covert the space URL Encoding into a '+' and create a hash map.
+         * @param users Strings with user's login, password and URN with
+         *  space characters as separators. Each of login, password and urn
+         *  are URL-encoded substrings.
+         * @return Map from login/password pairs to URNs.
+         */
+        private static Map<String, String> convert(final String... users) {
```
Comment (mkordas) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @sebing method should be named as a noun

File: src/main/java/org/takes/facets/auth/PsBasic.java [hunk 3]
```diff
@@ -226,26 +233,9 @@ public Fake(final boolean cond) {
...
-            this.usernames = new HashMap<String, String>(users.length);
-            for (final String user : users) {
-                final String unified = user.replace("%20", "+");
-                PsBasic.Default.validateUser(unified);
-                this.usernames.put(
-                    PsBasic.Default.key(unified),
-                    unified.substring(unified.lastIndexOf(' ') + 1)
-                );
-            }
+            this.usernames = convert(users);
```
Comment (mkordas) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @sebing can't we lazlily calculate it in `urn`?

File: src/main/java/org/takes/facets/auth/PsBasic.java [hunk 4]
```diff
@@ -269,6 +259,27 @@ public Default(final String... users) {
         }
 
         /**
+         * Covert the space URL Encoding into a '+' and create a hash map.
+         * @param users Strings with user's login, password and URN with
```
Comment (mkordas) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @sebing shouldn't it be `users'` as it is for many users?

File: src/main/java/org/takes/facets/auth/PsBasic.java [hunk 5]
```diff
@@ -269,6 +259,27 @@ public Default(final String... users) {
...
+         *  space characters as separators. Each of login, password and urn
```
Comment (mkordas) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @sebing `urn` -> `URN`

**Comment by mkordas** [comment]
@sebing please try fixing these things, I'll finish review tomorrow

File: src/main/java/org/takes/facets/auth/PsBasic.java (hunk 1 above)
Comment (sebing) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @mkordas done.

File: src/main/java/org/takes/facets/auth/PsBasic.java (hunk 2 above)
Comment (sebing) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @mkordas done

File: src/main/java/org/takes/facets/auth/PsBasic.java (hunk 5 above)
Comment (sebing) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @mkordas done

File: src/main/java/org/takes/facets/auth/PsBasic.java (hunk 4 above)
Comment (sebing) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @mkordas done

File: src/main/java/org/takes/facets/auth/PsBasic.java (hunk 3 above)
Comment (sebing) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @mkordas I feel loading them in the constructor is right for this time.
- We are storing the users for entering. (alternatively we have to store the URN to store)
- When we enter, we are entering the user not the URN.
This is my understanding. :)

**Comment by sebing** [comment]
@mkordas can you re-review the code?

**Comment by mkordas** [comment]
@sebing the code is good, we should try assigning you to the issue again

File: src/main/java/org/takes/facets/auth/PsBasic.java (hunk 3 above)
Comment (mkordas) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @sebing right, but that's a bit against http://www.yegor256.com/2015/05/07/ctors-must-be-code-free.html

Comment (sebing) [inline src/main/java/org/takes/facets/auth/PsBasic.java]: @mkordas we have done the initialization of a field the same way here.
(https://github.com/yegor256/takes/blob/master/src/main/java/org/takes/facets/auth/codecs/CcAES.java#L92)

**Comment by mkordas** [comment]
@sebing unfortunately @davvd assigned issue to someone else, so we need to close this pull request. Please try to do issues below 10 days in the future, otherwise you may lose assignment anytime.

**Comment by mkordas** [comment]
@sebing can you close this PR?

**Comment by sebing** [comment]
@mkordas Closing the PR as this task is taken by another pr(https://github.com/yegor256/takes/pull/640)

**Comment by mkordas** [comment]
@sebing thanks

**Comment by davvd** [comment]
@ypshenychka please, let us know what do you think about this ticket, according to our [QA rules](http://at.teamed.io/qa.html)

**Comment by ypshenychka** [comment]
@mkordas 
According to our [QA Rules](http://at.teamed.io/qa.html):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

There were only two design issues found during code review.
Please confirm that you'll try to find at least three major problems while future reviews along with cosmetic complaints.

**Comment by mkordas** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@mkordas Thanks

**Comment by ypshenychka** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@ypshenychka thanks for the QA review, we'll work better next time

**Comment by davvd** [comment]
@mkordas I added 10 mins to @ypshenychka (for QA review) in transaction 79370471

Many thanks! **28 mins** were added to your account in Transaction ID `56dbf509468d035a720000a1` (task took 76 hours and 57 mins)

extra minutes for review comments (c=13)

+28 to your rating, your total score is [+8160](http://www.netbout.com/b/36080?open=rating)

**Pull request closed without merging**